### ISSUE TITLE
Trailing space in list of notification email addresses causes SMTP server to reject outbound notification emails

### ISSUE THREAD

**Issue by axis80** [verbb/formie#1739]
### Describe the bug

Our outbound SMTP server (Zoho Mail) was rejecting Email Notifications with the message "553 Recipient domain not specified."  I looked at the configuration for the form and found there was a trailing space at the end of the list of email addresses.  I suggest you run trim() on that field's value whenever the setting is saved, and also before sending emails.

Not sure if the behavior is the same with other SMTP servers.  Probably easier to just trim() it and call it a day, than to investigate.

### Steps to reproduce

1. Configure Craft to send mail through Zoho Mail's SMTP server.
2. Create an Email Notification for a form.
3. Enter a list of email addresses separated by commas, and include a trailing space character.
4. Submit the form.
5. View the queue logs and you will see that delivery of the notification failed, with the "553 Recipient domain not specified" response from the outbound SMTP server.

### Form settings

- Multi-page form: No
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: No

### Craft CMS version

Craft Pro 4.7.0

### Plugin version

2.1.3

### Multi-site?

Yes

**Comment by engram-design** [comment]
That's odd, because that's [exactly](https://github.com/verbb/formie/blob/9902b1b58bf0112868eb11887de8392d3e303a44/src/services/Emails.php#L542) what we do, and I can confirm on my end reproducing that there's no spaces in emails. I wonder how that's getting through on your end?

You're just statically providing email recipients, correct? Not using field values?

**Comment by axis80** [comment]
Yep, we're just providing a static list of three email addresses.  IIRC, here's how all this happened:

- business admin person told me that form notifications were failing for one form
- I looked in logs and found they were failing due to SMTP error "553 Recipient domain not specified." 
- I looked at a failed job in the queue manager in the Craft web UI, and noticed the space in the JSON (or maybe I noticed it in a log file, I'm not sure).
- I opened up the form in Formie, opened up that particular email notification, and sure enough, there was a space at the end of the comma-separated list of email addresses.
- I removed the space, went back to the queue manager screen, and was able to resend all failed jobs

Since I resent them all they are no longer viewable from the queue manager, but I did find this bit of JSON in the log file **queue-2024-02-27.log**:

![image](https://github.com/verbb/formie/assets/6088477/0c09ae02-599d-4f1b-98ba-34c4f4529a47)

We just recently upgraded to Craft 4 (and upgraded our Formie plugin in the process) in the last couple weeks.  Is it possible that value might have been saved in a previous version of Formie that didn't use trim()?

**Comment by engram-design** [comment]
That's really odd. I don't mind about saving it with a space, but the sanitizing before sending is what that code above references. 

Unless of course this submission was made on the Craft 3 version of Formie, where there might've been an issue with `trim()`, and it's sat there in a failed state, until you've tried again in Formie 2 for Craft 4. Maybe unlikely, but just wanted to check.

Still, that shouldn't affect the processing of emails as it's sanitized before sending.

I'll investigate further!
